### ISSUE TITLE
Add search form

### ISSUE THREAD

**Issue by ZHB** [meons/HEVS-Android-tourism#8]
We can have a search form for easily find tourist by his reference. Practical when there will be a lot of data.

**Comment by maidmaid** [comment]
:+1:, and why not adding a find by date of creation (cf #11) ?

**Comment by ZHB** [comment]
I just added a simple search form by tourist reference in the base template. I have however some questions : 
- At the moment I have one controller for the form. When form si submitted, I render a template for search result. Is it better to redirect to a "search_result" route with tourists list parameters ? But is it possible ?
- I created a TouristType and nested TouristSearchByReference type for the form creation. But I read some people making a mapping entity class (so they don't have values without null values in their entity). What do you think about it ?
- Can you review that and make some comments ? 478b0b58c7362d542c49ae9b0990f70e36088fc5

Thanks

P.S.
Yes, searching by date is a good idea.

**Comment by meons** [comment]
For the first one I think render a template it's for search result is good. Why would you redirect to a route and pass trough a controller one more time ?
For the second one I have no idea, I never gone so far with Symfony forms =/

**Comment by maidmaid** [comment]
- I think that we can _merge_ your `searchByReferenceAction` in `indexAction`. So `indexAction` will contain the search form and the list tourist (or the filtered list if we have submitted search form).
- Currently, I see that TouristType is [not really used](https://github.com/ZHB/HEVS-Android-tourism/search?utf8=%E2%9C%93&q=TouristType) (just for inheritance). And I don't see where use it, because for our case, we would just perform a search in tourist. Why not just create a TouristSearchType without inheritance and containing all for the search ?
- Yes, you can create an `AppBundle\Model\TouristSearch` class and use it for your search form.

**Comment by ZHB** [comment]
I created TouristType on case we create an advance search form with dates and references. 

Ok for the maping class, I will change it.

Yes, we can let indexAction do the work. But I think in indexAction it is better to only show new tourists ?

**Comment by maidmaid** [comment]
I prefer say me that indexAction is able to list all but by default, it show new tourist. Or, the easiest, to list all, order by `is_new` flag and creation date. So, new tourists are always in first place. Are you ok with me ?

**Comment by ZHB** [comment]
Yes that's OK for me. But I don't think I will have the time to do that before tommorow and after that I leave to Croatia until 20 of august. 

Simeon or Dany can you do that changes ?

**Comment by maidmaid** [comment]
For 1.0 milestone, we have to hide this search form.
